core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -166,6 +166,10 @@ class KafkaController(val config: KafkaConfig,
   @volatile private var offlinePartitionCount = 0
   @volatile private var preferredReplicaImbalanceCount = 0
   @volatile private var globalTopicCount = 0
+  // https://docs.google.com/document/d/1s9_klx0OFZBf8-Hw87ZaXWPPqs1kf0EEgAckz4n0mzc/edit#heading=h.j3hmr1fm6kkj
+  // shows that each topic has a 4 bytes overhead when they are counted toward the 1M zk jute.maxbuffer limit.
```

Comment:
You could optionally mention that zkCli.sh prints the total size as part of its exception when it fails on > 1 MB responses.  (Or just for your own info.)

Reply:
Thank you, that's good to know.